flexmeasures/api/v3_0/sensors.py

```diff
@@ -154,3 +189,347 @@ def get_data(self, response: dict):
         :status 422: UNPROCESSABLE_ENTITY
         """
         return json.dumps(response)
+
+    @route("/<id>/schedules/trigger", methods=["POST"])
+    @use_kwargs(
+        {"sensor": SensorIdField(data_key="id")},
+        location="path",
+    )
+    @use_kwargs(
+        {
+            "soc_sensor_id": fields.Str(data_key="soc-sensor", required=False),
+            "roundtrip_efficiency": QuantityField(
+                "%", validate=validate.Range(min=0, max=1)
+            ),
+            "value": fields.Float(),  # todo: in the CLI equivalent, this field is named 'soc-at-start'
+            "soc-min": fields.Float(),  # todo: in postUdiEvent, this field was named 'soc_min'
+            "soc-max": fields.Float(),  # todo: in postUdiEvent, this field was named 'soc_max'
+            "start_of_schedule": AwareDateTimeField(
+                data_key="datetime", format="iso", required=False
+            ),  # todo: in the CLI equivalent, the data key for this field is named 'from'
+            "unit": fields.Str(
+                validate=OneOf(
+                    [
+                        "kWh",
+                        "MWh",
+                    ]
+                ),
+            ),  # todo: allow unit to be set per field, using QuantityField("%", validate=validate.Range(min=0, max=1))
+            "targets": fields.List(fields.Nested(TargetSchema)),
+        },
+        location="json",
```

Comment:
Maybe add a TODO about allowing a `duration` parameter, instead of falling back to `FLEXMEASURES_PLANNING_HORIZON`?